pkg/build/api/v1/conversion.go

```diff
@@ -114,6 +114,32 @@ func convert_v1_BuildTriggerPolicy_To_api_BuildTriggerPolicy(in *BuildTriggerPol
 	return nil
 }
 
+func convert_v1_BuildConfigSpec_To_api_BuildConfigSpec(in *BuildConfigSpec, out *newer.BuildConfigSpec, s conversion.Scope) error {
+	if err := s.DefaultConvert(in, out, conversion.DestFromSource); err != nil {
+		return err
+	}
+	// Remove unrecognized trigger types
+	triggers := []newer.BuildTriggerPolicy{}
+	for _, t := range out.Triggers {
```

Comment:
Yeah, PrepareForCreate/Update is fine.

On Fri, Sep 11, 2015 at 3:07 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/build/api/v1/conversion.go
> <https://github.com/openshift/origin/pull/4635#discussion_r39305887>:
>
> > @@ -114,6 +114,32 @@ func convert_v1_BuildTriggerPolicy_To_api_BuildTriggerPolicy(in *BuildTriggerPol
> >  	return nil
> >  }
> >
> > +func convert_v1_BuildConfigSpec_To_api_BuildConfigSpec(in *BuildConfigSpec, out *newer.BuildConfigSpec, s conversion.Scope) error {
> > +	if err := s.DefaultConvert(in, out, conversion.DestFromSource); err != nil {
> > +		return err
> > +	}
> > +	// Remove unrecognized trigger types
> > +	triggers := []newer.BuildTriggerPolicy{}
> > +	for _, t := range out.Triggers {
>
> clean up in PrepareForCreate/PrepareForUpdate
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4635/files#r39305887>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift